plugins/available/pyenv.plugin.bash

```diff
@@ -1,12 +1,34 @@
+# shellcheck shell=bash
 cite about-plugin
 about-plugin 'load pyenv, if you are using it'
 
-export PYENV_ROOT="$HOME/.pyenv"
-pathmunge "$PYENV_ROOT/bin"
+# https://github.com/pyenv/pyenv
 
-[[ `which pyenv 2>/dev/null` ]] && eval "$(pyenv init - bash)"
+# Load after basher
+# BASH_IT_LOAD_PRIORITY: 260
 
-#Load pyenv virtualenv if the virtualenv plugin is installed.
+# Don't modify the environment if we can't find the tool:
+# - Check if in $PATH already
+# - Check if installed manually to $PYENV_ROOT
+# - Check if installed manually to $HOME
+_command_exists pyenv \
+	|| [[ -n "$PYENV_ROOT" && -x "$PYENV_ROOT/bin/pyenv" ]] \
+	|| [[ -x "$HOME/.pyenv/bin/pyenv" ]] \
+	|| return 0
+
+# Set PYENV_ROOT, if not already set
+export PYENV_ROOT="${PYENV_ROOT:-$HOME/.pyenv}"
+
+# Add PYENV_ROOT/bin to PATH, if that's where it's installed
+if ! _command_exists pyenv && [[ -x "$PYENV_ROOT/bin/pyenv" ]]; then
```

Comment:
the first check can not fail, as we check in line 14 that the command does not exist

Reply:
@NoahGorny because the line following. This if statement is not checking for the condition to exit upon, it's checking whether it should pathmunge. While maybe the ordering of statements could be adjusted for efficiency, I'd rather leave that for a different PR.